cmd/juju/waitfor/application.go

```diff
@@ -164,31 +166,34 @@ func (c *applicationCommand) waitFor(ctx ScopeContext) func(string, []params.Del
 				if entityInfo.Application == name {
 					c.units[entityInfo.Name] = entityInfo
 				}
-			}
-		}
 
-		if !c.found {
-			logger.Infof("application %q not found, waiting...", name)
-			return false, nil
+			case *params.MachineInfo:
+				if delta.Removed {
+					delete(c.machines, entityInfo.Id)
+					break
+				}
+				c.machines[entityInfo.Id] = entityInfo
+			}
 		}
 
-		currentStatus := c.appInfo.Status.Current
-		logOutput := currentStatus.String() != "unset" && len(c.units) > 0
+		var currentStatus status.Status
+		if c.appInfo != nil {
+			currentStatus = c.appInfo.Status.Current
+			c.appInfo.Status.Current = deriveApplicationStatus(currentStatus, c.units)
 
-		appInfo := c.appInfo
-		appInfo.Status.Current = deriveApplicationStatus(currentStatus, c.units)
-
-		scope := MakeApplicationScope(ctx, &appInfo)
-		if done, err := runQuery(q, scope); err != nil {
-			return false, errors.Trace(err)
-		} else if done {
-			return true, nil
-		}
+			if found, err := run(q); err != nil {
+				return false, errors.Trace(err)
+			} else if found {
+				return true, nil
+			}
 
-		if logOutput {
-			logger.Infof("application %q found with %q, waiting for goal state", name, currentStatus)
+			logger.Infof("application %q found with %q, waiting...", name, currentStatus)
```

Comment:
The capitalization of the found and not found Infof messages is inconsistent.  Marking here, but happens with application, unit, model etc. 

```
$ juju wait-for model --query="len(machines) > 3" default
model "default" found, waiting...
Model "default" is running
```

Reply:
Good shout.